### ISSUE TITLE
sig_unblock: lacks unblocking of segv

### ISSUE THREAD

**Issue by xhebox** [troglobit/finit#51]
Open a new issue since it's not the same issue! Comment on issue50 has been deleted.

I infer from this that you suppose to block SIGSEGV only on services. But actually sig_unblock did not unblock SIGSEGV, though you commented that it will unblock all signals when starting children.
```
 	uev_signal_init(ctx, &sigchld_watcher, sigchld_cb, NULL, SIGCHLD);
 
 	/* Trap SIGSEGV in case service callbacks crash */
	uev_signal_init(ctx, &sigsegv_watcher, sigsegv_cb, NULL, SIGSEGV);
 
 	/* Stopping init is a bit tricky. */
 	uev_signal_init(ctx, &sigstop_watcher, sigstop_cb, NULL, SIGSTOP);
```

And this is my temporary patch.
```
--- sig.c	2016-11-27 04:05:25.281957561 +0000
+++ b/sig.c	2016-11-27 04:05:25.281957561 +0000
@@ -355,6 +355,7 @@
 	sigaddset(&nmask, SIGTERM);
 	sigaddset(&nmask, SIGUSR1);
 	sigaddset(&nmask, SIGUSR2);
+	sigaddset(&nmask, SIGSEGV);
 	sigprocmask(SIG_UNBLOCK, &nmask, NULL);
 
 	/* Reset signal handlers that were set by the parent process */
--- service.c	2016-11-27 04:05:25.281957561 +0000
+++ b/service.c	2016-11-27 04:05:25.281957561 +0000
@@ -143,7 +143,7 @@
 {
 	int i, result = 0;
 	pid_t pid;
-	sigset_t nmask, omask;
+	sigset_t nmask, omask, segvmask;
 
 	if (!svc)
 		return 1;
@@ -190,6 +190,9 @@
 	sigaddset(&nmask, SIGCHLD);
 	sigprocmask(SIG_BLOCK, &nmask, &omask);
 
+	sigemptyset(&segvmask);
+	sigaddset(&segvmask, SIGSEGV);
+
 	pid = fork();
 	if (pid == 0) {
 		int status;
@@ -260,6 +263,8 @@
 
 				/* Reset signals */
 				sig_unblock();
+				/* Reblock segv */
+				sigprocmask(SIG_BLOCK, &segvmask, NULL);
 
 				execlp("logger", "logger", "-t", svc->cmd, "-p", "daemon.info", NULL);
 				_exit(0);
@@ -280,6 +285,8 @@
 		}
 	logger_err:
 		sig_unblock();
+		/* Reblock segv */
+		sigprocmask(SIG_BLOCK, &segvmask, NULL);
 
 		if (svc->inetd.cmd)
 			status = svc->inetd.cmd(svc->inetd.type);
```

**Comment by xhebox** [comment]
other signals may need to be reviewed :D

**Comment by troglobit** [comment]
I've removed the `SIGSEGV` handling in 829ff0b, since it is no longer needed, and audited the signals used in Finit vs what is unblocked before each child is forked-off.

Hopefully things work better for you now.  Let me know otherwise and I'll reopen this issue.
